sdk/src/cose_validator.rs

```diff
@@ -642,6 +642,123 @@ fn get_sign_certs(sign1: &coset::CoseSign1) -> Result<Vec<Vec<u8>>> {
     get_unprotected_header_certs(sign1)
 }
 
+// get OCSP der
+fn get_ocsp_der(sign1: &coset::CoseSign1) -> Option<Vec<u8>> {
+    if let Some(der) = sign1
+        .unprotected
+        .rest
+        .iter()
+        .find_map(|x: &(Label, Value)| {
+            if x.0 == Label::Text("rVals".to_string()) {
+                Some(x.1.clone())
+            } else {
+                None
+            }
+        })
+    {
+        match der {
+            Value::Map(rvals_map) => {
+                // find OCSP value if available
+                rvals_map.iter().find_map(|x: &(Value, Value)| {
+                    if x.0 == Value::Text("ocspVals".to_string()) {
+                        match &x.1 {
+                            Value::Array(ocsp_responses) => {
+                                // our implementation will only check the first OCSP response
+                                if !ocsp_responses.is_empty() {
+                                    let ocsp_rsp_val = &ocsp_responses[0];
+
+                                    match ocsp_rsp_val {
+                                        Value::Bytes(ocsp_rsp_bytes) => {
+                                            Some(ocsp_rsp_bytes.clone())
+                                        }
+                                        _ => None,
+                                    }
+                                } else {
+                                    None
+                                }
+                            }
+                            _ => None,
+                        }
```

Comment:
This seems to sill pass the unit tests and reduces the lines of code a bit. You may be able to keep going up the else chain

```suggestion
                        x.1.as_array()
                            .and_then(|ocsp_rsp_val| ocsp_rsp_val.first())
                            .and_then(Value::as_bytes)
                            .cloned()
```

Reply:
Nice reduction.